quickjs_test.go

```diff
@@ -149,6 +150,25 @@ func TestJsFunction(t *testing.T) {
 	require.True(t, result.IsString() && result.String() == "args test")
 }
 
+func TestMemoryLimit(t *testing.T) {
+
+	runtime := NewRuntime()
+	defer runtime.Free()
+
+	const kB = 1 << 10
+	runtime.SetMemoryLimit(32 * kB)
```

Comment:
At a limit of 16 kB the `runtime.NewContext()` panics. I am not sure why that happens. It is not obvious to me what other methods might panic with a low memory limit as well.

I see several options how this could be handled:
* Document that a low limit could cause some methods to panic and make it the responsibility of the caller to set a limit that is safe for their use case.
    * This is not a nice experience for clients of the library.
    * It puts a low burden on the library maintainer.
* Silently enforce a minimal limit, ie set it to a minimal value when calling code tries to set it too low.
    * I would say that a safe limit is one that guarantees no method will panic because of it.
    * This only makes sense if we can determine what _is_ a safe minimal limit.
    * If it turns out that the minimal limit is higher than it needs to be, it can be safely lowered.
    * If it turns out that the minimal limit is too low, then the reaction is to either:
        * Raise it. This can cause some clients to unexpectedly use more memory than before.
        * Leave it as is. This defeats the purpose of having the limit.
        * Remove it. This can break code that was previously protected from panic by the minimal limit.
* Enforce a minimal limit either by panicking or returning an error.
    * This has the same difficulties with determining the limit as above.
    * Just as with silent enforcement, if the limit is determined wrong all solutions I see either defeat the purpose or break backwards compatibility:
        * Raise it. This can cause some programs that used to work to fail.
        * Leave it as is. This defeats the purpose of having the limit.
        * Remove it. This can break code that was previously protected from panic by the minimal limit.
* Commit to returning errors anywhere an operation might fail due to a low memory limit.
    * This does not require determining a safe limit.
    * This does require one of the following:
        * Determining with certainty which methods can fail under a low memory limit and adding an error return value to those.
            * If this is determined wrong, it defeats the purpose of doing it - error return values would need to be added to methods that did not have them before. This is not backwards compatible.
        * Preemptively adding an error return value to all methods.

I would say the "document + make user responsible for setting a safe limit" is probably the safest route to take.